### ISSUE TITLE
Support for OXEN <https://oxen.io> routing

### ISSUE THREAD

**Issue by Kreyren** [Gedsh/InviZible#90]

**Comment by majestrate** [comment]
do you mean lokinet?

**Comment by Kreyren** [comment]
> do you mean lokinet? @majestrate

Afaik it was rebranded from loki? O.o

**Comment by Yquas** [comment]
Adding Lokinet routing was already in issue #51. It is too much effort, at least as of now. And there are surely some things critisisable about Lokinet/Oxen that Tor hasn't.
I mean the site even wants me to do a G00gle reCaptcha... And this Oxen thingy is also a cryptocurrency and all over blockchains. Doesn't sound that great.

**Comment by majestrate** [comment]
> > do you mean lokinet? @majestrate
> 
> Afaik it was rebranded from loki? O.o

no it remains lokinet

**Comment by majestrate** [comment]
> there are surely some things critisisable about Lokinet/Oxen that Tor hasn't.

what are these things specifically?

**Comment by Yquas** [comment]
Oh I see you're working on it. I'm quiet then. The Session community was not nice at all when seeing criticism.
I hate blockchains wasting energy. And there were some people saying developers of Lokinet are connected with far right people. It's much speculation, I know. But it's not even the topic here. Let's see what Gedsh will answer.

**Comment by majestrate** [comment]
> Oh I see you're working on it. I'm quiet then. The Session community was not nice at all when seeing criticism.

when others criticize your favorite project it's easy to get emotional.

> I hate blockchains wasting energy. 

the coin integration is optional and doesn't use any PoW at all at this point. it's all full proof of stake.

> And there were some people saying developers of Lokinet are connected with far right people. 

This is a flat out lie and some people have even suggested we explore a libel case in germany (we may have a solid enough case), I personally don't think it's worth anyone's time to sue people over internet rumors as it's absolutely petty.

> It's much speculation, I know. But it's not even the topic here. Let's see what Gedsh will answer.

We have a jni glue layer for android VPN api and are working on a flutter application that wraps it. The jni bindings would probably be the best way to use lokinet on android.

**Comment by Kreyren** [comment]
> Oh I see you're working on it. I'm quiet then. @Yquas

You've already raised concerns so lets discuss them in a civilized manner! ^-^

If you have something beyond hearsay that can be proven beyond reasonable doubt, then i will be happy to listen and reconsider my support of OXEN as needed.

**Comment by Gedsh** [comment]
I do not plan to add any more anonymous networks to InviZible, as this can significantly complicate InviZible and its configuration.

Personally, I am wary of anonymous networks that use blockchain technology, since the blockchain stores its state for years, and perhaps in the future this can be used to de-anonymize users. May be OXEN is using blockchain in a secure way, but I don't see the benefits that InviZible already has with Tor, DNSCrypt and I2P.

**Comment by majestrate** [comment]
> I do not plan to add any more anonymous networks to InviZible, as this can significantly complicate InviZible and its configuration.

this is perfectly reasonable. I can see it getting in the way of tor a bit as well.

> 
> Personally, I am wary of anonymous networks that use blockchain technology, since the blockchain stores its state for years, and perhaps in the future this can be used to de-anonymize users. May be OXEN is using blockchain in a secure way, but I don't see the benefits that InviZible already has with Tor, DNSCrypt and I2P.
> 

yeah i would agree partially with this, the blockchain parts are also contentious for me personally however over time they have shown to be providing what it advertised adequately (a directory authority atop a blockchain). I wish there was enough interest in an opennet network fork with none of the coin parts but not enough interest has materialized in an altruism based network when the chance to run a node at a fairly decent profit margin exists. The opennet mode of operation is fully supported and always will be there just isn't a network for it right now from the aforementioned lack of interest from potential relay operators.

> > The jni bindings would probably be the best way to use lokinet on android.
> 
> I agree!

the hardest part for me/us is UI so integration in a project like this would definitely be a net positive on our end, on your end I'd have to further consider what would make it worth your while to include lokinet. so far I don't have much to offer other than an IP tunnel that onion routes stuff like tor does for tcp streams but with IP packets, which is still pretty cool I guess.

**Comment by Gedsh** [comment]
>  don't have much to offer other than an IP tunnel that onion routes stuff like tor does for tcp streams but with IP packets, which is still pretty cool I guess.

InviZible includes I2P that can handle any type of traffic. But this network is still not very popular.

**Comment by majestrate** [comment]
> > don't have much to offer other than an IP tunnel that onion routes stuff like tor does for tcp streams but with IP packets, which is still pretty cool I guess.
> 
> InviZible includes I2P that can handle any type of traffic. But this network is still not very popular.

True, but I2P is soooooo slow in comparison to lokinet, (we need to put out a point release to fix a regression in lokinet clients as the network took a bit of a hit with 0.9.0). I2P's way of doing explicit app integration to make applications explicitly i2p aware has been one of the biggest barriers of entry for I2P's adoption, even when you the most gung-ho developers that want to add i2p to their application you have to do it manually. lokinet working at the IP layer with DNS triggered address mapping makes this porting process as minimal as technically possible if it is needed at all, most applications I have used don't need porting of any kind. I originally was planning on adding an IP/DNS auto mapping layer to i2pd but it ended up being less work to rewrite everything from zero and make a new network and protocol than to make i2p's protocols play nice with that mode of operation. :<

**Comment by Gedsh** [comment]
> True, but I2P is soooooo slow in comparison to lokinet

All anonymous networks have their own advantages and disadvantages. There is no anonymous technology that is definitely better than others. Unfortunately InviZible cannot include all of them. I do not want to say that the lokinet is bad. But like other networks, it has its own advantages and disadvantages. I think it needs a separate application

**Comment by Kreyren** [comment]
What about some kind of "plugin" management that would allow outsourcing of the lokinet in invizable to oxen then?

**Comment by Gedsh** [comment]
There is no easy way to create plugins in Android. It is usually much more complex than direct implementation.

**Comment by FrozzDay** [comment]
sad to see that lokinet android unmaintained
